activerecord/lib/active_record/persistence.rb

```diff
@@ -758,7 +758,8 @@ def verify_readonly_attribute(name)
 
     def _raise_record_not_destroyed
       @_association_destroy_exception ||= nil
-      raise @_association_destroy_exception || RecordNotDestroyed.new("Failed to destroy the record", self)
+      message = self.errors.first.try(:second) || "Failed to destroy the record"
```

Comment:
Not liking `self.errors.first.try(:second)`. `self.` should be removed. `try` should be replaced with `&.`. `first&.second` doesn't seem very clear to me.

Although I haven't looked at this code before, but isn't `@_association_destroy_exception` partly here to facilitate this?

Reply:
if `@ _association_destroy_exception` can be used, i would just populate that in `destroy!` if errors are present.

Reply:
I also checked codebase to find `try()` usage and be consistent but if i could use `&.` it'd be great.

Reply:
Where should I populate `@ _association_destroy_exception`?

https://github.com/rails/rails/blob/2a470d73a75ebf8cd7975e469bd82586d9234442/activerecord/lib/active_record/callbacks.rb#L306